internal/mode/static/state/graph/nginxproxy.go

```diff
@@ -0,0 +1,30 @@
+package graph
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+)
+
+func getNginxProxyConfig(
+	nps map[types.NamespacedName]*ngfAPI.NginxProxy,
+	gc *v1beta1.GatewayClass,
+) *ngfAPI.NginxProxy {
+	var cfg *ngfAPI.NginxProxy
```

Comment:
can we simplify here?
(1) construct nsName from the Parameter Ref (if it exists)
(2) look up in the map

Reply:
Yep, I like that.